### PR TITLE
Deprecate SimpleFileCollection

### PR DESCRIPTION
**Pull request by acourtneybrown**

### CODE REVIEW

File: subprojects/core/src/integTest/groovy/org/gradle/api/file/ProjectLayoutIntegrationTest.groovy [hunk 1]
```diff
@@ -217,7 +217,7 @@ class ProjectLayoutIntegrationTest extends AbstractIntegrationSpec {
         'FileCollection'             | 'Closure'        | 'project.layout.filesFor({ "%s/src/resource/file.txt" })'
         'FileCollection'             | 'List'           | 'project.layout.filesFor([ "%s/src/resource/file.txt" ])'
         'FileCollection'             | 'array'          | 'project.layout.filesFor([ "%s/src/resource/file.txt" ] as Object[])'
-        'FileCollection'             | 'FileCollection' | 'project.layout.filesFor(new org.gradle.api.internal.file.collections.SimpleFileCollection(new File("%s/src/resource/file.txt")))'
+        'FileCollection'             | 'FileCollection' | 'project.layout.filesFor(new org.gradle.api.internal.file.collections.FileCollectionAdapter(new org.gradle.api.internal.file.collections.ListBackedFileSet(new File("%s/src/resource/file.txt"))))'
```
Comment (lptr) [inline subprojects/core/src/integTest/groovy/org/gradle/api/file/ProjectLayoutIntegrationTest.groovy]: Why not use `ImmutableFileCollection.of()` here?

Comment (acourtneybrown) [reply subprojects/core/src/integTest/groovy/org/gradle/api/file/ProjectLayoutIntegrationTest.groovy]: good point.  i was in the mindset from updating the `ImmutableFileCollectionTest` to *not* use `ImmutableFileCollection`, but is perfectly reasonable here 😃

**Comment by acourtneybrown** [comment]
@lptr PR should be ready for you

File: subprojects/core/src/integTest/groovy/org/gradle/api/file/ProjectLayoutIntegrationTest.groovy [hunk 2]
```diff
@@ -217,7 +217,7 @@ class ProjectLayoutIntegrationTest extends AbstractIntegrationSpec {
         'FileCollection'             | 'Closure'        | 'project.layout.filesFor({ "%s/src/resource/file.txt" })'
         'FileCollection'             | 'List'           | 'project.layout.filesFor([ "%s/src/resource/file.txt" ])'
         'FileCollection'             | 'array'          | 'project.layout.filesFor([ "%s/src/resource/file.txt" ] as Object[])'
-        'FileCollection'             | 'FileCollection' | 'project.layout.filesFor(new org.gradle.api.internal.file.collections.SimpleFileCollection(new File("%s/src/resource/file.txt")))'
+        'FileCollection'             | 'FileCollection' | 'project.layout.filesFor(org.gradle.api.internal.file.collections.ImmutableFileCollection.of(new File("%s/src/resource/file.txt")))'
```
Comment (lptr) [inline subprojects/core/src/integTest/groovy/org/gradle/api/file/ProjectLayoutIntegrationTest.groovy]: Nit: you could replace `org.gradle.api.internal.file.collections.ImmutableFileCollection` with `${ImmutableFileCollection.name}` to keep this shorter and refactor-safe.

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -39,6 +39,11 @@ The following are the newly deprecated items in this Gradle release. If you have
 - `FileCollection.add()` is now deprecated. Use `ConfigurableFileCollection.from()` instead. You can create a `ConfigurableFileCollection` via `Project.files()`.
 - `FileCollection.stopExecutionIfEmpty()` is deprecated without a replacement. You can use `@SkipWhenEmpty` on a `FileCollection` property, or throw a `StopExecutionException` in your code manually instead.
 
+### `SimpleFileCollection`
+
+The internal `SimpleFileCollection` implementation of `FileCollection` has been deprecated.
+You should use `ProjectLayout.filesFor(Object...)` if you need an immutable `FileCollection` or `ProjectLayout.mutableFilesFor(Object...)` if you need a mutable `ConfigurableFileCollection`.
```
Comment (lptr) [inline subprojects/docs/src/docs/release/notes.md]: Let's use `Project.files()` here, and don't mention immutable file collections yet.

File: subprojects/core/src/integTest/groovy/org/gradle/api/file/ProjectLayoutIntegrationTest.groovy (hunk 2 above)
Comment (acourtneybrown) [reply subprojects/core/src/integTest/groovy/org/gradle/api/file/ProjectLayoutIntegrationTest.groovy]: good idea.  done.

File: subprojects/docs/src/docs/release/notes.md (hunk 3 above)
Comment (acourtneybrown) [reply subprojects/docs/src/docs/release/notes.md]: done.

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -39,6 +39,11 @@ The following are the newly deprecated items in this Gradle release. If you have
 - `FileCollection.add()` is now deprecated. Use `ConfigurableFileCollection.from()` instead. You can create a `ConfigurableFileCollection` via `Project.files()`.
 - `FileCollection.stopExecutionIfEmpty()` is deprecated without a replacement. You can use `@SkipWhenEmpty` on a `FileCollection` property, or throw a `StopExecutionException` in your code manually instead.
 
+### `SimpleFileCollection`
+
+The internal `SimpleFileCollection` implementation of `FileCollection` has been deprecated.
+You should use `Project.files()` instead.
```
Comment (lptr) [inline subprojects/docs/src/docs/release/notes.md]: I really hate this, but I guess we should use `Project.files()` in the deprecation message, too. :/

Comment (acourtneybrown) [reply subprojects/docs/src/docs/release/notes.md]: done ... sadly :(

**Pull request merged**